Automattic/jetpack
# CRM: Cleanup after deleting custom fields

## Proposed changes:
* This PR eliminates orphaned references to custom fields within field sorting settings when removing custom fields.

### Other information:


## Jetpack product discussion
https://github.com/Automattic/zero-bs-crm/issues/3057

## Does this pull request change what data or activity we track or use?
No.

## Testing instructions:

1. In settings, create a new address custom field. (`wp-admin/admin.php?page=zerobscrm-plugin-settings&tab=customfields`)
2. In settings, go to field sorts and click save. (`/wp-admin/admin.php?page=zerobscrm-plugin-settings&tab=fieldsorts`).
3. Go back to the previous settings page and delete the newly created custom field.
4. Go back to the field sorts page, you should already see a php warning.
![image](https://user-images.githubusercontent.com/37049295/231500143-3f4ea1f8-da1e-41c4-9d0b-75bd72782ee3.png)
![image](https://user-images.githubusercontent.com/37049295/231500192-431a8962-774e-4753-980c-21c1ea87eba1.png)
5. Go to the edit page for any contact. (e.g. `/wp-admin/admin.php?page=zbs-add-edit&action=edit&zbstype=contact&zbsid=1`)
6. Click Update Contact.
7. You should see more warnings.
![image](https://user-images.githubusercontent.com/37049295/231500561-6a0285fd-3ee0-4912-be6b-d4fb8e6295cf.png)
8. Do the same for fields where `Hidden` was checked in `/wp-admin/admin.php?page=zerobscrm-plugin-settings&tab=fieldsorts`

In `fix/crm-3057-cleanup-after-deleting-custom-fields` none of this happen.
